src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -348,18 +348,33 @@ internal static ImmutableArray<SyntaxReference> GetDeclaringSyntaxReferenceHelpe
             foreach (Location location in locations)
             {
                 // Location may be null. See https://github.com/dotnet/roslyn/issues/28862.
-                if (location == null)
+                if (location == null || !location.IsInSource)
                 {
                     continue;
                 }
-                if (location.IsInSource)
+
+                if (location.SourceSpan.Length != 0)
                 {
-                    SyntaxToken token = (SyntaxToken)location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start);
+                    SyntaxToken token = location.SourceTree.GetRoot().FindToken(location.SourceSpan.Start);
                     if (token.Kind() != SyntaxKind.None)
                     {
                         CSharpSyntaxNode node = token.Parent.FirstAncestorOrSelf<TNode>();
                         if (node != null)
+                        {
                             builder.Add(node.GetReference());
+                        }
+                    }
+                }
+                else
+                {
+                    SyntaxNode parent = location.SourceTree.GetRoot().FindNode(location.SourceSpan);
```

Comment:
>SyntaxNode parent = location.SourceTree.GetRoot().FindNode(location.SourceSpan); [](start = 20, length = 80)

Could you please add a comment describing what special situation we are trying to handle here and how do we intend to do this?

Reply:
Added a comment and also used a more expensive but general approach in case similar problem arises in the future.

---
In reply to: [459843952](https://github.com/dotnet/roslyn/pull/46225#discussion_r459843952) [](ancestors = 459843952)